### DISCUSSION TITLE
Customizable Stateful Action (Win Arrow Keys)

### DISCUSSION THREAD

**Discussion by b0mbard1er** [rxhanson/RectanglePro-Community#92] [Ideas]
First off, thank you for the great app. Moving from Windows to Mac was 100 times easier using rectangle pro for window management. 

I find myself using the  "Stateful" Win Arrow Keys action a lot, but I would like to request a new feature for it. Currently, it seems like it cycles between halves of the screen(s) and restoring to original position.

It would be great if we could customize the cycle actions-- for instance, to display in thirds instead of halves, and skip over the original position when cycling through. Or even to potentially skip a display or change behavior on that display, though that might be a challenge to implement.

Thanks again, and I hope you consider implementing a bit of customization for this great feature!

**Comment by rxhanson** [comment]
Thanks for the feedback.

You can build your own repeated actions with the custom shortcut builder, as described in this community tutorial: https://github.com/rxhanson/RectanglePro-Community/discussions/5

Unfortunately, this isn't stateful like the win arrow keys at the moment - the state machine is only based on the previous execution of the same shortcut and not on any other shortcuts. I do have plans to add stateful behavior to it, but it will likely be a while before I get to it.
